### PR TITLE
feat: add React text

### PR DESCRIPTION
**Pull request by MrtinoRG**
## Summary by Sourcery

Enhance the appendix with detailed information on tool-augmented models, specifically ReAct-style environments, and their performance. Update the documentation to include explanations of the tools used and their impact on model performance.

Enhancements:
- Enhance the appendix to include detailed information on tool-augmented models, specifically ReAct-style environments, and their performance.

Documentation:
- Update the appendix to provide a comprehensive explanation of tool-augmented models and their performance, including the use of specific tools like WolframAlpha and ArXiv API.

### CODE REVIEW

File: src/tex/appendix.tex [hunk 1]
```diff
@@ -343,7 +343,7 @@ \subsection{Model performance} \label{sec:model_performance_app}
 However, since some subjects are composed of questions from different sources, the ranking of the models is, in some instances, different from the one on \chembenchmini.
 
 \begin{table}
-    \caption{\textbf{Performance of the models on the \chembench corpus.} The table shows the fraction of questions answered correctly by the models for different skills and difficulty levels.}
+    \caption{\textbf{Performance of the models on the \chembench corpus.} The table shows the fraction of questions answered correctly by the models for different skills and difficulty levels. Models with \enquote{T-one} in the name were run for Temperature=1, which allow to study the temperature effect in the benchmark. Systems with \enquote{ReAct} in the name are tool augmented, i.e., they can call external tools such as web search or a calculator to better answer the questions. However, we limit those systems to a maximum of ten calls to the LLM. This constraint led the systems to often not find the correct answer within the specified number of calls. In this case, we consider the answer as incorrect (see \Cref{sec:react-environment}).}
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:346]:
```suggestion
    \caption{\textbf{Performance of the models on the \chembench corpus.} The table shows the fraction of questions answered correctly by the models for different skills and difficulty levels. Models with \enquote{T-one} in the name were run for a temperature of 1, which allows us to study the temperature effect in the benchmark. Systems with \enquote{ReAct} in the name are tool augmented, i.e., they can call external tools such as web search or a calculator to answer the questions better. However, we limit those systems to ten calls to the \gls{llm}. This constraint often led the systems to not find the correct answer within the specified number of calls. In this case, we consider the answer as incorrect (see \Cref{sec:react-environment}).}
```

File: src/tex/appendix.tex [hunk 2]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
     \label{fig:tool_use}
 \end{figure}
 
+\clearpage
+\subsection{Tool augmented models} \label{sec:react-environment}
+In addition to directly prompting LLMs, we also investigated the performance of tool-augmented systems.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:592]:
```suggestion
In addition to directly prompting \glspl{llm}, we also investigated the performance of tool-augmented systems.
```

File: src/tex/appendix.tex [hunk 3]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+For this, we investigated the two models which showed the best overall results, \GPTFourO and \ClaudeThreeFiveSonnet (\oone overperformed both models but is not recommended using this model with such \enquote{reasoning} prompts such as ReAct\autocite{yao2022react} or CoT\autocite{wei2023cot}).
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:593]:
```suggestion
For this, we investigated the two models which showed the best overall results, \GPTFourO and \ClaudeThreeFiveSonnet (\oone overperformed both models but is not recommended using this model with such \enquote{reasoning} prompts such as ReAct\autocite{yao2022react} or \gls{cot}\autocite{wei2023cot}). 
```

File: src/tex/appendix.tex [hunk 4]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
     \label{fig:tool_use}
 \end{figure}
 
+\clearpage
+\subsection{Tool augmented models} \label{sec:react-environment}
+In addition to directly prompting LLMs, we also investigated the performance of tool-augmented systems.
+For this, we investigated the two models which showed the best overall results, \GPTFourO and \ClaudeThreeFiveSonnet (\oone overperformed both models but is not recommended using this model with such \enquote{reasoning} prompts such as ReAct\autocite{yao2022react} or CoT\autocite{wei2023cot}). 
+For both models, we created a ReAct-style tool augmentation environment in which models had access to WolframAlpha, the ArXiv API, the Wikipedia access, and a web search (using Brave browser API).
+We based the election of these tools based in the most used tools by the humans (see \Cref{fig:tool_use}).
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:595]:
```suggestion
We based the selection of these tools on the most used tools by humans (see \Cref{fig:tool_use}).
```

File: src/tex/appendix.tex [hunk 5]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+Additionally, we added two specific tools to convert IUPAC names to SMILES, and SMILES to IUPAC names.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:596]:
```suggestion
Additionally, we added two specific tools to convert \gls{iupac} names to \gls{smiles}, and \gls{smiles} to \gls{iupac} names.
```

File: src/tex/appendix.tex [hunk 6]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+These conversion tools allow us to better understand how the agents perform for some specific questions in this agent environment configurations.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:597]:
```suggestion
These conversion tools allow us to better understand how the agents perform for some specific questions in this agent environment configuration.
```

File: src/tex/appendix.tex [hunk 7]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+We implemented the systems using Langchain with the default ReAct prompt and constrained the system to a maximum of ten LLM calls.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:598]:
```suggestion
We implemented the systems using Langchain with the default ReAct prompt and constrained the system to a maximum of ten \gls{llm} calls.
```

is there a good citation for langchain?

File: src/tex/appendix.tex [hunk 8]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+
+While for \ClaudeThreeFiveSonnet the overall performance increased (less than 1\%), they show a sloped decrease for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}).
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:600]:
```suggestion
While for \ClaudeThreeFiveSonnet the overall performance increased (by less than 1\%), we observe a decrease in performance for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}). 
```

File: src/tex/appendix.tex (hunk 7 above)
Comment (AdrianM0) [reply src/tex/appendix.tex]: their git repo: https://github.com/langchain-ai/langchain/blob/master/CITATION.cff

File: src/tex/appendix.tex [hunk 9]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+For both models, we created a ReAct-style tool augmentation environment in which models had access to WolframAlpha, the ArXiv API, the Wikipedia access, and a web search (using Brave browser API).
+We based the election of these tools based in the most used tools by the humans (see \Cref{fig:tool_use}).
+Additionally, we added two specific tools to convert IUPAC names to SMILES, and SMILES to IUPAC names.
+These conversion tools allow us to better understand how the agents perform for some specific questions in this agent environment configurations.
+We implemented the systems using Langchain with the default ReAct prompt and constrained the system to a maximum of ten LLM calls.
+
+While for \ClaudeThreeFiveSonnet the overall performance increased (less than 1\%), they show a sloped decrease for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}). 
+If we study the results by each type of question, we observe an improvement for questions regarding electron count or point group questions.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:601]:
```suggestion
If we study the results by each type of question, we observe an improvement for questions regarding electron counts or point groups of compounds.
```

File: src/tex/appendix.tex [hunk 10]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+On the other hand, the scores decreased question asking about the number of isomers or information contained in pictograms.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:602]:
```suggestion
However, the scores decreased for questions asking about the number of isomers or \gls{ghs} pictograms.
```

there was no "on the one hand"

File: src/tex/appendix.tex [hunk 11]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+For the specific questions about converting IUPAC names to SMILES, the results decreased notably despite the specific tools that were prepared for those questions.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:603]:
```suggestion
For the specific questions about converting \gls{iupac} names to \gls{smiles}, the results decreased notably despite the models having access to specific tools prepared for those questions. 
```

File: src/tex/appendix.tex (hunk 8 above)
Comment (AdrianM0) [reply src/tex/appendix.tex:600]: @MrtinoRG as far as I see there is no improvement for Claude? it is almost the same number, without the agent being slightly better?

File: src/tex/ms.tex [hunk 12]
```diff
@@ -249,7 +249,7 @@ \section{Conclusions}
 Our work shows that carefully curated benchmarks can provide a more nuanced understanding of the capabilities of \glspl{llm} in the chemical sciences.
 Importantly, our findings also illustrate that more focus is required in developing better human-model interaction frameworks, given that models cannot estimate their limitations.
 
-While our findings indicate many areas for further improvement of \gls{llm}-based systems, it is also important to realize that clearly defined metrics have been the key to the progress of many fields of \gls{ml}, such as computer vision.
+While our findings indicate many areas for further improvement of \gls{llm}-based systems, such as for the ReAct systems (more discussion in \cref{sec:react-environment}), it is also important to realize that clearly defined metrics have been the key to the progress of many fields of \gls{ml}, such as computer vision.
```
Comment (kjappelbaum) [inline src/tex/ms.tex:252]:
```suggestion
While our findings indicate many areas for further improvement of \gls{llm}-based systems, such as for agents (more discussion in \cref{sec:react-environment}), it is also important to realize that clearly defined metrics have been the key to the progress of many fields of \gls{ml}, such as computer vision.
```

Comment (kjappelbaum) [reply src/tex/ms.tex:252]: I'd tend to keep the language here simple - in the end, it is a chemistry journal for the average chemist :)

File: src/tex/appendix.tex [hunk 13]
```diff
@@ -588,6 +587,24 @@ \subsection{Human baseline} \label{sec:human_baseline}
...
+For both models, we created a ReAct-style tool augmentation environment in which models had access to WolframAlpha, the ArXiv \gls{api}, Wikipedia, and web search (using Brave search \gls{api}).
+We based the selection of these tools on the most used tools by humans (see \Cref{fig:tool_use}).
+Additionally, we added two specific tools to convert \gls{iupac} names to \gls{smiles}, and \gls{smiles} to \gls{iupac} names.
+These conversion tools allow us to better understand how the agents perform for some specific questions in this agent environment configuration.
+We implemented the systems using Langchain with the default ReAct prompt and constrained the system to a maximum of ten \gls{llm} calls.
+
+While for \ClaudeThreeFiveSonnet the overall performance increased (by less than 1\%), we observe a decrease in performance for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}).
```
Comment (AdrianM0) [inline src/tex/appendix.tex:600]:
```suggestion
While for \ClaudeThreeFiveSonnet the overall performance did not change, we observe a decrease in performance for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}). 
```

File: src/tex/ms.tex [hunk 14]
```diff
@@ -249,7 +249,7 @@ \section{Conclusions}
 Our work shows that carefully curated benchmarks can provide a more nuanced understanding of the capabilities of \glspl{llm} in the chemical sciences.
 Importantly, our findings also illustrate that more focus is required in developing better human-model interaction frameworks, given that models cannot estimate their limitations.
 
-While our findings indicate many areas for further improvement of \gls{llm}-based systems, it is also important to realize that clearly defined metrics have been the key to the progress of many fields of \gls{ml}, such as computer vision.
+While our findings indicate many areas for further improvement of \gls{llm}-based systems, such as for agents (more discussion in \cref{sec:react-environment}), it is also important to realize that clearly defined metrics have been the key to the progress of many fields of \gls{ml}, such as computer vision.
```
Comment (kjappelbaum) [inline src/tex/ms.tex:252]:
```suggestion
While our findings indicate many areas for further improvement of \gls{llm}-based systems, such as for agents (more discussion in \Cref{sec:react-environment}), it is also important to realize that clearly defined metrics have been the key to the progress of many fields of \gls{ml}, such as computer vision.
```

**Comment by kjappelbaum** [comment]
If we could add a citation to langchain at a good place we are done here, I think!

```
@software{Chase_LangChain_2022,
author = {Chase, Harrison},
month = oct,
title = {{LangChain}},
url = {https://github.com/langchain-ai/langchain},
year = {2022}
}
```

**Comment by MrtinoRG** [comment]
Done!

File: src/tex/references.bib [hunk 15]
```diff
@@ -291,6 +291,14 @@ @Article{caufield2023structured
   year      = {2024},
 }
 
+@software{Chase_LangChain_2022,
```
Comment (kjappelbaum) [inline src/tex/references.bib:294]: need to later check how this renders, but it is good for now. Thanks!

**Review by kjappelbaum — APPROVED** [review]
wohoo! 🚀

**Pull request merged**